bson/bsonrw/extjson_reader.go

```diff
@@ -225,7 +230,7 @@ func (ejvr *extJSONValueReader) Type() bsontype.Type {
 
 func (ejvr *extJSONValueReader) Skip() error {
 	switch ejvr.stack[ejvr.frame].mode {
-	case mElement, mValue:
+	case mElement, mValue, mArray:
```

Comment:
Per offline discussion, I added a test for skipping an array alongside the embedded document test. `extjson_reader#skipArray()` was actually working just fine, but for some reason this case in `Skip` (which calls `skipArray()`) was missing `mArray` and was throwing the error below when an array was skipped. Should be working now.